src/sync-client.js

```
@@ -314,8 +314,20 @@ var self = {
     });
   },
 
-
+  /**
+   * Allows to override default method for checking if application is online
+   *
+   * @param handler - function that checks if network is available
+   */
+  setNetworkStatusHandler: function(handler){
```

probably add a check here to make sure `handler` is actually a function

---

Fail fast and verify inputs is a good call